File: Insure-e/Insure-e/src/main/resources/static/js/app/InsuranceService.js
'use strict'

angular.module('insure.services', []).factory('InsuranceService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getAllModules = function() {
				return $http.get(CONSTANTS.getAllModules);
			}
			service.getAllCoverage = function() {
				return $http.get(CONSTANTS.getAllCoverage);
			}
			service.caculateTariff = function(tariffDto) {
				return $http.post(CONSTANTS.calculateTariff, tariffDto);
			}
			service.saveTariff = function(tariffDto) {
				return $http.post(CONSTANTS.saveTariff, tariffDto);
			}
			service.getAllTariff = function() {
				return $http.get(CONSTANTS.getAllTariff);
			}
			return service;
		} ]);

File: Insure-e/Insure-e/src/main/java/com/insurance/Insuree/service/InsuranceService.java
package com.insurance.Insuree.service;

import java.util.List;

import com.insurance.Insuree.dto.CoverageDto;
import com.insurance.Insuree.dto.ModuleDto;
import com.insurance.Insuree.dto.TariffDto;

public interface InsuranceService {

	List<ModuleDto> getAllModule();

	List<CoverageDto> getAllCoverage();

	public TariffDto calculateTariff(TariffCalculatorService tariffCalculatorService, TariffDto tariffDto);

	public TariffDto estimateTariffCalculations(TariffDto tariffDto);

	public void saveTariff(TariffDto tariffDto);

	List<TariffDto> getAllTariff();
}
